Update auto_random docs (#7475)
pingcap/docs: auto-random.md

> 

Reviewer: Above, at Line 70:
```
In the above `CREATE TABLE` statement, `3` shard bits are specified. The range of the number of shard bits is `[1, field_max_bits)`. `field_max_bits` is the length of bits occupied by the primary key column.
```

Is probably not fully correct:

```
tidb> create table t (a bigint auto_random(62) primary key, b varchar(255));
ERROR 8216 (HY000): Invalid auto random: max allowed auto_random shard bits is 15, but got 62 on column `a`
```

Author: https://github.com/pingcap/tidb/pull/17119 added:

```go
// MaxAutoRandomBits is the max value of auto sharding.
const MaxAutoRandomBits = 15
```

Author: 5fd5ff499c4e730f9404b79d42642431f8ea7eaf

Pay attention to the following restrictions when you use `AUTO_RANDOM`:

> - Specify this attribute for the primary key column **ONLY** of `bigint` type. Otherwise, an error might occur. In addition, when the attribute of the primary key is `NONCLUSTERED`, `AUTO_RANDOM` is not supported even on the integer primary key. For more details about the primary key of the `CLUSTERED` type, refer to [clustered index](/clustered-indexes.md).

Reviewer: ```suggestion
- Specify this attribute for the primary key column **ONLY** of `bigint` type. Otherwise an error occur. In addition, when the attribute of the primary key is `NONCLUSTERED`, `AUTO_RANDOM` is not supported even on the integer primary key. For more details about the primary key of the `CLUSTERED` type, refer to [clustered index](/clustered-indexes.md).
```

- You cannot use `ALTER TABLE` to modify the `AUTO_RANDOM` attribute, including adding or removing this attribute.

> - You cannot use `ALTER TABLE` to changing from `AUTO_INCREMENT` to `AUTO_RANDOM` if the maximum value is close to the maximum value of the column type.

Reviewer: I have not tested, but I think the technical issue is when the maximum value is close to use the shard bit, see https://github.com/pingcap/tidb/pull/20512#discussion_r593060069 so the more shard bits used, the lower the max auto_increment.

Author: Is this ok as is or do I need to change the doc?

Reviewer 2: ```suggestion
- You cannot use `ALTER TABLE` to change a column from `AUTO_INCREMENT` to `AUTO_RANDOM` if the maximum value of the column is close to the maximum value of the column type.
```

Reviewer: I think this is generic enough for the documentation.

- You cannot change the column type of the primary key column that is specified with `AUTO_RANDOM` attribute.
- You cannot specify `AUTO_RANDOM` and `AUTO_INCREMENT` for the same column at the same time.

> - You cannot specify `AUTO_RANDOM` and `DEFAULT` (the default value of a column) for the same column at the same time.

Reviewer: Maybe something like:
- When using `AUTO_RANDOM` it is hard to go back to `AUTO_INCREMENT` due to very high auto generated values.

Author: Like this?
```suggestion
- You cannot specify `AUTO_RANDOM` and `DEFAULT` (the default value of a column) for the same column at the same time.
- When using `AUTO_RANDOM` it is hard to go back to `AUTO_INCREMENT` due to very high auto generated values.
```

Reviewer: Yes, LGTM :)

Pay attention to the following restrictions when you use `AUTO_RANDOM`:

> - Specify this attribute for the primary key column **ONLY** of `bigint` type. Otherwise an error occur. In addition, when the attribute of the primary key is `NONCLUSTERED`, `AUTO_RANDOM` is not supported even on the integer primary key. For more details about the primary key of the `CLUSTERED` type, refer to [clustered index](/clustered-indexes.md).

Reviewer 2: ```suggestion
- Specify this attribute for the primary key column **ONLY** of `bigint` type. Otherwise, an error occurs. In addition, when the attribute of the primary key is `NONCLUSTERED`, `AUTO_RANDOM` is not supported even on the integer primary key. For more details about the primary key of the `CLUSTERED` type, refer to [clustered index](/clustered-indexes.md).
```

- You cannot use `ALTER TABLE` to modify the `AUTO_RANDOM` attribute, including adding or removing this attribute.
- You cannot use `ALTER TABLE` to changing from `AUTO_INCREMENT` to `AUTO_RANDOM` if the maximum value is close to the maximum value of the column type.
- You cannot change the column type of the primary key column that is specified with `AUTO_RANDOM` attribute.
- You cannot specify `AUTO_RANDOM` and `AUTO_INCREMENT` for the same column at the same time.
- You cannot specify `AUTO_RANDOM` and `DEFAULT` (the default value of a column) for the same column at the same time.

> - When using `AUTO_RANDOM` it is hard to go back to `AUTO_INCREMENT` due to very high auto generated values.

Reviewer 2: ```suggestion
- When`AUTO_RANDOM` is used on a column, it is difficult to change the column attribute back to `AUTO_INCREMENT` because the auto-generated values might be very large.
```
